### DISCUSSION TITLE
threadsafe in publisher and subscriber operations.

### DISCUSSION THREAD

**Discussion by qclzdh** [eclipse-iceoryx/iceoryx#1009] [Q&A]
I have 3 questions about the publisher and subscriber opteration is threadsafe or not.
1. It seems UntypedPublisherImpl::load, UntypedPublisherImpl::publish and UntypedPublisherImpl::release is not threadsafe, right?
2. If i have two publisher in one process,  is UntypedPublisherImpl_a::load and UntypedPublisherImpl_b::publish threadsafe?
3. If i have one publisher and one subscriber in one process, is UntypedSubscriberImpl::release and UntypedPublisherImpl::load threadsafe?

**Comment by elBoberido** [comment]
Accessing publisher and subscriber is not thread safe. If you need to access one publisher/subscriber from multiple threads you have to use a mutex to guard the access.

1. Yes, accessing one publisher in multiple threads is not safe
2. Yes, two publisher in different threads with access to each of them only in their thread is safe
3. Yes, a publisher and subscriber in different threads with access to each of them only in their thread is safe
